Wrappers/Python/cil/io/TIFF.py

```diff
@@ -29,40 +29,97 @@
 import glob
 import re
 import numpy as np
+from cil.io import utilities
+import json
 
 import logging
 
 logger = logging.getLogger(__name__)
 
+def compress_data(data, scale, offset, dtype):
+    '''Compress data to dtype using scale and offset
+    
+    Parameters
+    ----------
+    data : numpy array
+    scale : float
+    offset : float
+    dtype : numpy dtype
+    
+    returns compressed casted data'''
+    if dtype == data.dtype:
+        return data
+    if data.ndim > 2:
+        # compress each slice
+        tmp = np.empty(data.shape, dtype=dtype)
+        for i in range(data.shape[0]):
+            tmp[i] = compress_data(data[i], scale, offset, dtype)
+    else:
+        tmp = data * scale + offset
+        tmp = tmp.astype(dtype)
+    return tmp
+
+def save_scale_offset(fname, scale, offset):
+    '''Save scale and offset to file
+    
+    Parameters
+    ----------
+    fname : string
+    scale : float
+    offset : float
+    '''
+    dirname = os.path.dirname(fname)
+    txt = os.path.join(dirname, 'scaleoffset.json')
+    d = {'scale': scale, 'offset': offset}
+    utilities.save_dict_to_file(txt, d)
+
 class TIFFWriter(object):
     '''Write a DataSet to disk as a TIFF file or stack'''
     
     def __init__(self,
                  **kwargs):
         '''
-        :param data: the data to save to TIFF file(s)
-        :type data: DataContainer, AcquisitionData or ImageData
-        :param file_name: string defining the file name prefix
-        :type file_name: string
-        :param counter_offset: int indicating at which number the index starts. 
-          For instance, if you have to save 10 files the index would by default go from 0 to 9.
-          By counter_offset you can offset the index: from `counter_offset` to `9+counter_offset`
-        :type counter_offset: int, default 0
+
+        Parameters
+        ----------
+        data : DataContainer, AcquisitionData or ImageData
+            This represents the data to save to TIFF file(s)
+        file_name : string
+            This defines the file name prefix, i.e. the file name without the extension.
+        counter_offset : int, default 0.
+            counter_offset indicates at which number the ordinal index should start.
+            For instance, if you have to save 10 files the index would by default go from 0 to 9.
+            By counter_offset you can offset the index: from `counter_offset` to `9+counter_offset`
+        compression : str, default None. Accepted values None, 'uint8', 'uint16'
```

Comment:
yay! Can you open an issue for nexus writer and reader too, and add it to the 23.0.0 milestone.

Reply:
See #1408